src/quest/historian.py

```diff
@@ -832,6 +832,7 @@ async def _func(*a, **kw):
                     assert record['task_id'] == task_id
 
             logging.debug(f'Completing task {task_id}')
+            self._history.clear()
```

Comment:
We wanted the Workflow Manager to have control of the cleaning process, but the test we talked about didn't include one. Also, should the cleaning be as close to the history as possible? Is there a case where there isn't a workflow manager?

Reply:
After adding this a few tests are breaking. Again, not sure if this is the correct implementation.